src/natcap/invest/habitat_quality.py

```diff
@@ -64,128 +60,116 @@
             "name": "Future Land Cover"
         },
         "lulc_bas_path": {
-            "type": "raster",
+            **utils.LULC_ARG,
+            **utils.PROJECTED,
             "required": False,
-            "validation_options": {
-                "projected": True,
-            },
             "about": (
                 "Optional.  A GDAL-supported raster file.  If the baseline "
                 "LULC is provided, rarity outputs will be created for the "
-                "current and future LULC. The baseline LULC can have it's "
-                "own threat rasters (optional), where each threat raster "
-                "file path is defined in the <b>Threats Data</b> CSV. "
-                "If there are no threat rasters and the threat paths are "
-                "left blank in the CSV column, degradation and habitat "
-                "quality outputs will not be generated for the baseline "
-                "LULC.<br/><br/> "
-                "Each cell should represent a LULC code as an Integer.  The "
-                "dataset should be in a projection where the units are in "
-                "meters and the projection used should be defined. The LULC "
-                "codes must match the codes in the Sensitivity table.  If "
-                "possible the baseline map should refer to a time when "
-                "intensive management of the landscape was relatively rare."),
+                "current and future LULC. The baseline LULC can have it's own "
+                "threat rasters (optional), where each threat raster file "
+                "path is defined in the <b>Threats Data</b> CSV. If there are "
+                "no threat rasters and the threat paths are left blank in the "
+                "CSV column, degradation and habitat quality outputs will not "
+                "be generated for the baseline LULC.<br/><br/> Each cell "
+                "should represent a LULC code as an Integer.  The dataset "
+                "should be in a projection where the units are in meters and "
+                "the projection used should be defined. The LULC codes must "
+                "match the codes in the Sensitivity table.  If possible the "
+                "baseline map should refer to a time when intensive "
+                "management of the landscape was relatively rare."),
             "name": "Baseline Land Cover"
         },
         "threats_table_path": {
-            "validation_options": {
-                "required_fields": [
-                    "THREAT", "MAX_DIST", "WEIGHT", "DECAY", "CUR_PATH"],
-            },
             "type": "csv",
-            "required": True,
+            "columns": {
+                "threat": {"type": "freestyle_string"},
+                "max_dist": {
+                    "type": "number",
+                    "units": u.kilometer,
```

Comment:
Yes, this input is kilometers people! So many convolutions busted with inputs of `5000`m. Anyway, I love that units are formally being specified up front.

Reply:
Yay! I'm glad these changes are satisfying to people besides me 😀